x/manifest/keeper/msg_server_test.go

```diff
@@ -14,225 +14,101 @@ import (
-					Percentage: 100_000_000,
-				},
-			}, false, 0, "umfx"),
-			success: false,
+			name:   "fail; bad authority",
+			sender: acc.String(),
+			payouts: []types.PayoutPair{
+				types.NewPayoutPair(acc, "umfx", 1),
+			},
+			shouldFail: true,
 		},
 		{
-			desc:   "duplicate address",
-			sender: authority.String(),
-			p: types.NewParams([]*types.StakeHolders{
-				{
-					Address:    acc.String(),
-					Percentage: 50_000_000,
-				},
-				{
-					Address:    acc.String(),
-					Percentage: 50_000_000,
-				},
-			}, false, 0, "umfx"),
-			success: false,
+			name:   "fail; bad bech32 authority",
+			sender: "bad",
+			payouts: []types.PayoutPair{
+				types.NewPayoutPair(acc, "umfx", 1),
+			},
+			shouldFail: true,
 		},
 		{
-			desc:    "success none",
-			sender:  authority.String(),
-			p:       types.NewParams([]*types.StakeHolders{}, false, 0, "umfx"),
-			success: true,
+			name:   "fail; payout to bad address",
+			sender: authority.String(),
+			payouts: []types.PayoutPair{
+				types.NewPayoutPair(acc, "umfx", 1),
+				{Address: "badaddr", Coin: sdk.NewCoin("umfx", sdkmath.NewInt(2))},
+				types.NewPayoutPair(acc3, "umfx", 3),
+			},
+			shouldFail: true,
 		},
 		{
-			desc:   "success many stake holders",
+			name:   "fail; payout with a 0 token",
 			sender: authority.String(),
-			p: types.NewParams([]*types.StakeHolders{
-				{
-					Address:    acc.String(),
-					Percentage: 1_000_000,
-				},
-				{
-					Address:    acc2.String(),
-					Percentage: 1_000_000,
-				},
-				{
-					Address:    authority.String(),
-					Percentage: 98_000_000,
-				},
-			}, false, 0, "umfx"),
-			success: true,
+			payouts: []types.PayoutPair{
+				types.NewPayoutPair(acc, "umfx", 1),
+				types.NewPayoutPair(acc2, "umfx", 0),
+			},
+			shouldFail: true,
 		},
-	} {
-		tc := tc
-		t.Run(tc.desc, func(t *testing.T) {
-			// Set the params
-			_, err := ms.UpdateParams(f.Ctx, &types.MsgUpdateParams{
-				Authority: tc.sender,
-				Params:    tc.p,
-			})
-			require.Equal(t, tc.success, err == nil, err)
-
-			// Ensure they are set the same as the expected
-			if tc.success && len(tc.p.StakeHolders) > 0 {
-				params, err := f.App.ManifestKeeper.Params.Get(f.Ctx)
-				require.NoError(t, err)
-
-				require.Equal(t, tc.p.StakeHolders, params.StakeHolders)
-			}
-		})
 	}
-}
-
-func TestCalculatePayoutLogic(t *testing.T) {
-	_, _, authority := testdata.KeyTestPubAddr()
-	_, _, acc := testdata.KeyTestPubAddr()
-	_, _, acc2 := testdata.KeyTestPubAddr()
-	_, _, acc3 := testdata.KeyTestPubAddr()
-	_, _, acc4 := testdata.KeyTestPubAddr()
 
-	f := initFixture(t)
+	for _, c := range cases {
+		c := c
 
-	k := f.App.ManifestKeeper
+		t.Run(c.name, func(t *testing.T) {
+			payoutMsg := &types.MsgPayout{
+				Authority:   c.sender,
+				PayoutPairs: c.payouts,
+			}
 
-	k.SetAuthority(authority.String())
-	ms := keeper.NewMsgServerImpl(k)
+			_, err := ms.Payout(f.Ctx, payoutMsg)
+			if c.shouldFail {
+				require.Error(t, err)
+				return
```

Comment:
We should check the balances on failure to see if we didn't perform partial minting.

Reply:
ref: partial minting is not possible in the SDK - https://github.com/liftedinit/manifest-ledger/pull/68#discussion_r1624908622